engine/src/main/java/org/terasology/world/internal/EntityAwareWorldProvider.java

```diff
@@ -127,6 +127,23 @@ public Block setBlock(Vector3i pos, Block type) {
     }
 
     @Override
+    public Map<Vector3i, Block> setBlocks(Map<Vector3i, Block> blocks) {
+        if (GameThread.isCurrentThread()) {
+            Map<Vector3i, Block> oldTypes = super.setBlocks(blocks);
```

Comment:
Here, the old/replaced blocks are required to update the entites. With #2467 we argued that that `setBlock` should return info on success/failure, but not more. Turning `EntityAwareWorldProvider` into a normal system listening for de-/activation of block components would be compliant to the new interface.